whitepaper/wlroots.tex

```diff
@@ -0,0 +1,440 @@
+    \item \nameref{wlr input device}
+    \item \nameref{wlr keyboard}
+    \item \nameref{wlr output}
+    \item \nameref{wlr pointer}
+    \item \nameref{wlr tablet pad}
+    \item \nameref{wlr tablet tool}
+    \item \nameref{wlr touch}
+\end{itemize}
+
+\subsection{Wayland Interfaces}\label{wl interfaces}
+
+These types implement a specific interface in the (or \emph{a}) Wayland
+protocol. They implement much of the boilerplate, any useful state handling,
+raise events when necessary, and offer users functions for manipulating their
+resources. Note that there is some overlap with convenience types here. Many
+resoruces, such as \textbf{wlr_output}, provide functionality for both the
+compositor's benefit and the Wayland client.
+
+\begin{itemize}
+    \itemsep0em
+    \item \nameref{wlr compositor}
+    \item \nameref{wlr data device}
+    \item \nameref{wlr keyboard}
+    \item \nameref{wlr output}
+    \item \nameref{wlr pointer}
+    \item \nameref{wlr region}
+    \item \nameref{wlr screenshooter}
+    \item \nameref{wlr seat}
+    \item \nameref{wlr server decoration}
+    \item \nameref{wlr surface}
+    \item \nameref{wlr tablet pad}
+    \item \nameref{wlr tablet tool}
+    \item \nameref{wlr touch}
+    \item \nameref{wlr wl shell}
+    \item \nameref{wlr xdg shell v6}
+\end{itemize}
+
+\subsection{Special Types}\label{special types}
+
+Some types are spread throughout their respective subsystems and are not
+included in \code{include/types/*.h}. These include:
+
+\begin{itemize}
+    \itemsep0em
+    \item \nameref{wlr backend}
+    \item \nameref{wlr renderer}
+    \item \nameref{wlr texture}
+    \item \nameref{wlr xcursor}
+    \item \nameref{wlr xwayland}
+\end{itemize}
+
+\newpage
+\section{Type Reference}\label{type reference}
+
+\subsection{wlr_backend}\label{wlr backend}
+
+\code{wlr/backend.h}\\
+
+This type abstracts the functionality of hardware backends. For an overview of
+this system and available backends see \nameref{high level backends}.
+
+\subsection{wlr_box}\label{wlr box}
+
+\code{wlr/types/wlr_box.h}\\
+
+\textbf{wlr_box} is a simple data structure:
+
+\begin{verbatim}
+struct wlr_box {
+	int x, y;
+	int width, height;
+};
+\end{verbatim}
+
+A number of utility functions are available in the header.
+
+\subsection{wlr_compositor}\label{wlr compositor}
+
+TODO
+
+\subsection{wlr_cursor}\label{wlr cursor}
+
+TODO
+
+\subsection{wlr_data_device}\label{wlr data device}
+
+TODO
+
+\subsection{wlr_gamma_control}\label{wlr gamma control}
+
+TODO
+
+\subsection{wlr_input_device}\label{wlr input device}
+
+TODO
+
+\subsection{wlr_keyboard}\label{wlr keyboard}
+
+TODO
+
+\subsection{wlr_list}\label{wlr list}
+
+TODO
+
+\subsection{wlr_matrix}\label{wlr matrix}
+
+TODO
+
+\subsection{wlr_output}\label{wlr output}
+
+\code{wlr/types/wlr_output.h}\\
+
+\textbf{wlr_output} abstracts access to display hardware provided by backends.
+The most important functionality offered by \textbf{wlr_outputs} is the frame
```

Comment:
move the `s` out of the bold?

Reply:
I tried that initially and I don't think it really works.

Reply:
You can reword it so you don't have the s. Eg:  "the most important functionality provided by a wlr_output is... "

Reply:
Nah, this is going to come up all over the place and I'd rather not try to worm our way around it everywhere.

Reply:
Good point, best to just keep it like it is. I don't think anyone will be confused by the s, and it looks better being a part of the formatting.